contracts/ChipRegistry.sol

```diff
@@ -137,225 +124,224 @@ contract ChipRegistry is IChipRegistry, ClaimedPBT, Ownable {
-     * @param _projectClaimDataUri       URI pointing to location of off-chain data required to claim chips
-     */
-    function updateProjectMerkleRoot(
-        IProjectRegistrar _projectRegistrar,
-        bytes32 _merkleRoot,
-        string calldata _projectClaimDataUri
-    )
-        external
-    {
-        require(msg.sender == projectEnrollments[_projectRegistrar].projectPublicKey, "Caller must be project public key");
-        require(projectEnrollments[_projectRegistrar].creationTimestamp + 30 days > block.timestamp, "Update period has elapsed");
-        require(!projectEnrollments[_projectRegistrar].claimsStarted, "Claims have already started");
-
-        projectEnrollments[_projectRegistrar].merkleRoot = _merkleRoot;
-        projectEnrollments[_projectRegistrar].projectClaimDataUri = _projectClaimDataUri;
-        
-        emit ProjectMerkleRootUpdated(address(_projectRegistrar), _merkleRoot, _projectClaimDataUri);
-    }
-
-    /**
-     * @notice Allow a user to claim a chip from a project enrollment. Enrollment allows the chip to resolve to the project's preferred
+     * @notice Allow a project to add chips. Enrollment allows the chip to resolve to the project's preferred
      * service. Additionally, claiming creates a Physically-Bound Token representation of the chip.
      *
-     * @dev This function will revert if the chip has already been claimed, if invalid certificate data is provided or if the chip is
+     * @dev This function will revert if the chip has already been added, if invalid certificate data is provided or if the chip is
      * not part of the project enrollment (not in the project merkle root). Addtionally, there are checks to ensure that the calling
      * ProjectRegistrar has implemented the correct ERS logic. This function is EIP-1271 compatible and can be used to verify chip
      * claims tied to an account contract.
      *
      * @param _chipId                       Chip ID (address)
-     * @param _chipClaim                    Struct containing information for validating merkle proof, chip owner, and chip's ERS node
+     * @param _chipOwner                    Struct containing information for validating merkle proof, chip owner, and chip's ERS node
+     * @param _nameHash                     Label of the node in the ERS tree; typically the chipId unless the project wishes to use 
+     *                                      another unique identifier. The full ersNode will be used as the tokenId for the issued PBT.
      * @param _manufacturerValidation       Struct containing information for chip's inclusion in manufacturer's merkle tree
-     * @param _developerInclusionProof      Signature of the chipId signed by the project's public key
-     * @param _developerCustodyProof        Signature of the projectPublicKey signed by the chip's private key
      */
-    function claimChip(
+    
+    function addChip(
         address _chipId,
```

Comment:
One thing I don't love about this function is that we're reliant on the `ProjectRegistrar` to check for possession of the chip. Basically if someone were to get a list of chipId's included in a project before they've been added they could add the chips to their project and nothing could be done. Would recommend some sort of check like a signature here.

Reply:
I guess my fear is more that all it takes is someone getting their hands on a list of addresses in a prospective enrollment and then they could claim all those chips if the legitimate owners haven't already claimed. Feels like this completely severs on-chain chip ownership from possession of the chip. You basically have to tell all your customers that they need to safe guard the list of addresses or the whole operation could fail.